plugins/language_typst.lua

```diff
@@ -4,153 +4,213 @@ local syntax = require "core.syntax"
+
+		{ pattern = "0x[%dabcdef]+()%a*",          type = {"number", "keyword2"}          },
+		{ pattern = "0b[01]+()%a*",                type = {"number", "keyword2"}          },
+		{ pattern = "0o[01234567]+()%a*",          type = {"number", "keyword2"}          },
+		{ pattern = "%d+[.e]%d+()%a*",             type = {"number", "keyword2"}          },
+		{ pattern = "%d+()%a*",                    type = {"number", "keyword2"}          },
+
+		{ pattern = "%a%a+",                       type = "symbol"                        },
+
+	},
+
+
+	symbols = {
+		["set"]      = "keyword",
+		["let"]      = "keyword",
+		["show"]     = "keyword",
+		["include"]  = "keyword",
+		["import"]   = "keyword",
+		["if"]       = "keyword",
+		["else"]     = "keyword",
+		["for"]      = "keyword",
+		["in"]       = "keyword",
+		["while"]    = "keyword",
+		["break"]    = "keyword",
+		["continue"] = "keyword",
+		["return"]   = "keyword",
+		["as"]       = "keyword",
+		["or"]       = "keyword",
+		["and"]      = "keyword",
+		["not"]      = "keyword",
+
+		["true"]     = "literal",
+		["false"]    = "literal",
+	}
+}
 
-    },
+table.insert(typst_script.patterns, 1, { pattern = { "%(", "%)" }, syntax = typst_script })
+table.insert(typst_script.patterns, 1, { pattern = { "{",  "}"  }, syntax = typst_script })
+
+
+
+local typst_syntax = {
+	name = "Typst",
+	files = { "%.typ$" },
+	comment = "//",
+	block_comment = { "/*", "*/" },
+
+	patterns = {
+		{
+			pattern = { "#()import", "\n"    },
+			type    = { "literal", "keyword" },
+			syntax  = typst_script
+		},
+		{
+			pattern = { "#()include", "\n"   },
+			type    = { "literal", "keyword" },
+			syntax  = typst_script
+		},
+		{
+			pattern = { "#()set", "\n"       },
+			type    = { "literal", "keyword" },
+			syntax  = typst_script
+		},
+		{
+			pattern = { "#()let", "\n"       },
+			type    = { "literal", "keyword" },
+			syntax  = typst_script
+		},
+		{
+			pattern = { "#()show", "\n"      },
+			type    = { "literal", "keyword" },
+			syntax  = typst_script
+		},
+
+		{ pattern = { "#", "%s"},               type = "literal", syntax = typst_script },
+		{ pattern = { "%$", "%$", "\\" },       type = "literal", syntax = typst_math   },
+
+		{ pattern = { "```js", "```", "\\" },   type = "string", syntax = ".js"         },
+		{ pattern = { "```py", "```", "\\" },   type = "string", syntax = ".py"         },
+		-- Possibility to add more syntax later
+		{ pattern = { "`", "`", "\\" },         type = "string"                         },
+
+		{ pattern = "//.*",                     type = "comment"                        },
+		{ pattern = { "/%*", "%*/" },           type = "comment"                        },
+
+		--{ pattern = {"link"},                      type = "typst_underline"                }, -- add underline text later
+
+		-- Finish later
+		{ pattern = "^%s*=+%s.+%f[\n]",         type = "keyword"                        }, -- Bold
```

Comment:
I don't think you need to use `%f[\n]`:
```suggestion
		{ pattern = "^%s*=+%s.*",               type = "keyword"                        }, -- Bold
```